roles/README.md

```diff
@@ -0,0 +1,54 @@
+<h1 align="center">
+  Stratum Roles
+</h1>
+
+# Introduction
+
+We define roles for entities in the Stratum v2 Protocol Suite, which can be classified as either downstream or upstream in relation to each other.
+
+# List of Roles
+
+1 Mining Devices or Miners
+2 Pools
+3 Proxies
+    - Mining Proxy
+    - Translation Proxy
+4 Job Declarators
+    - Job Declarator Server (JDS)
+    - Job Declarator Client (JDC)
+
+
+## Mining Devices or Miners
+
+The term "miners" refers to the machines responsible for computing hashes. They come in various sizes and types, from large-scale corporate farms to smaller, decentralized setups. When discussing miners in relation to a pool, they are typically grouped based on how they communicate with the pool. For example, a mining farm with a hashing power of 10PH that communicates as a single unit with the pool is considered a single 'miner'. This is different from an individual operating a single S19 miner in their garage. Miners direct their hashing power towards a pool, as explained below. In Stratum v2, miners are considered the most downstream participants.
+
+## Pools
+
+Pools serve as intermediaries for coordinating hashing power and distributing mining rewards. They create tasks, validate blocks, and share them with the Bitcoin Network. Pools don't control hashing power but compete for it based on factors like latency and reliability. With Stratum v2, these factors can be significantly improved. Pools establish communication channels with downstream roles like Proxies or Mining Devices.
+
+## Proxies
+
+  Proxies are intermediaries between Miners and Pools that aggregate connections and translate mining communications from Sv1->Sv2 or Sv2->Sv1. Proxies may optionally provide additional functionality including monitoring services or job declaration optimizations. Both Miners and Pools can run Proxies, and they will do so for various reasons depending on the use-case. i
+
+  in addition, there are 2 proxies here on Stratum they include the following :
+
+  a. Mining Proxy  
+
+    The Sv2 Mining Proxy functions as a bridge connecting mining devices and the Sv2 Pool. It gathers mining requests from various devices, combines them, and sends them to the Sv2 Pool. It can establish group/extended channels with the upstream (Sv2 Pool) and standard channels with the downstream (Sv2 Mining Devices).
+
+  b. Translation Proxy
+
+    The Translator Proxy facilitates communication between Sv1 Mining Devices and an Sv2 Pool or Mining Proxy. It allows Sv1 devices to interface with Sv2-based mining infrastructure, bridging the protocol gap between Sv1 and Sv2. The Translator Proxy can establish extended channels with the upstream (Sv2 Pool or Mining Proxy). For instance, a Pool may employ a Translator Proxy as its initial connection service to accommodate both Sv1 and Sv2 connections. This involves establishing direct standard channels with Sv2 miners and using the proxy to translate messages with Sv1 miners.
```

Comment:
```suggestion
a. Mining Proxy  

The Sv2 Mining Proxy functions as a bridge connecting mining devices and the Sv2 Pool. It gathers mining requests from various devices, combines them, and sends them to the Sv2 Pool. It can establish group/extended channels with the upstream (Sv2 Pool) and standard channels with the downstream (Sv2 Mining Devices).

b. Translation Proxy

The Translator Proxy facilitates communication between Sv1 Mining Devices and an Sv2 Pool or Mining Proxy. It allows Sv1 devices to interface with Sv2-based mining infrastructure, bridging the protocol gap between Sv1 and Sv2. The Translator Proxy can establish extended channels with the upstream (Sv2 Pool or Mining Proxy). For instance, a Pool may employ a Translator Proxy as its initial connection service to accommodate both Sv1 and Sv2 connections. This involves establishing direct standard channels with Sv2 miners and using the proxy to translate messages with Sv1 miners.
```